serde/src/test/kotlin/com/amazon/ionhiveserde/objectinspectors/IonStructToMapObjectInspectorTest.kt

```diff
@@ -18,86 +18,482 @@ package com.amazon.ionhiveserde.objectinspectors
 import com.amazon.ion.IonStruct
 import com.amazon.ionhiveserde.ION
 import com.amazon.ionhiveserde.ionNull
+import com.amazon.ionhiveserde.objectinspectors.map.*
+import org.apache.hadoop.hive.common.type.HiveDecimal
 import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category.MAP
-import org.apache.hadoop.hive.serde2.objectinspector.primitive.PrimitiveObjectInspectorFactory
+import org.apache.hadoop.hive.serde2.objectinspector.ObjectInspector.Category.PRIMITIVE
 import org.junit.Test
+import java.sql.Date
+import java.sql.Timestamp
 import kotlin.test.assertEquals
 import kotlin.test.assertNull
+import kotlin.test.fail
 
 class IonStructToMapObjectInspectorTest {
-    private val valueElementInspector = com.amazon.ionhiveserde.objectinspectors.IonIntToIntObjectInspector(true)
-    private val subject = com.amazon.ionhiveserde.objectinspectors.IonStructToMapObjectInspector(valueElementInspector)
+    private val stringIntKeyElementInspector = IonFieldNameToStringObjectInspector()
```

Comment:
Thanks for all the work put into these tests :)

Do you think there's a way to reduce repetition here?

Reply:
Definitely - streamlined a bunch of the repetitive calls.